registry/remote/repository.go

```diff
@@ -126,6 +129,21 @@ func NewRepository(reference string) (*Repository, error) {
 	}, nil
 }
 
+func newRepositoryWithOptions(ref registry.Reference, opts *RepositoryOptions) (*Repository, error) {
```

Comment:
It would be better if we can have documentation on `newRepositoryWithOptions`.

Reply:
Done.